### ISSUE TITLE
GitHub plugin: credential lost after upgrade to GE v3

### ISSUE THREAD

**Issue by jrfnl** [gitextensions/gitextensions#6403]
## Current behaviour

Not sure if this was a hick-up or not, but after upgrading from GE2 to GE3, it seems that the OAuth token for the GitHub plugin has vanished.

## Expected behaviour

That previously set settings would be respected and remembered, even when upgrading to a new major version.

## Steps to reproduce

I would guess that the following steps would reproduce this, though I haven't tried:
1. Install GE2.
2. Add a GH OAuth token to the settings for the GH plugin.
3. Go to `GitHub` -> `Fork/Clone repository` and see the list of `My repositories` filling up (presuming you actually have some).
3. Upgrade to GE3.
4. Go to `GitHub` -> `Fork/Clone repository` and see the list of `My repositories` remain empty with a "no authentication details" notice.

## Screenshots

![GE-no-auth-details](https://user-images.githubusercontent.com/663378/54887832-964a7100-4e97-11e9-912e-5f53ce9f9252.png)

## Did this work in previous version of GitExtensions

I didn't experience this issue before after updating to new 2.x minor/patch releases.

## Environment

- Git Extensions 3.0.2.5232
- Build 23b6f51905006ccdde8701591df706284b4155dc
- Git 2.21.0.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3362.0
- DPI 96dpi (no scaling)

## Additional info

I probably "fixed" this in a round-a-bout way, i.e. going to the Settings page, seeing the Oauth key field empty, going to the GH website, creating an OAuth key via the Settings -> Developer Settings -> Personal Access Tokens page and pasting the new key into the settings field.
Only afterwards I realized that going via Plugins -> GitHub would get me an Oauth/browser screen to set it directly.

All the same, I figure I should mention that when using the above route (i.e. manually updating the key in settings), it won't take effect until GE has been restarted and it doesn't give any indication that a restart is needed.

**Comment by SAPHemlock** [comment]
Tracked down what happened here, and I do not think it should be "fixed" - explanation below.

**Why the token disappeared.** The plugin's token is stored under a key built as `<plugin prefix>` + `"OAuth Token"`, where the prefix used to be the plugin's description. In 2.51.05 the plugin declared `SetNameAndDescription("Github")`, so the key was `GithubOAuth Token`. Commit `bc052b7ea` "Fix GitHub capitalisation" (2018-07-25, shipped in v3.0) changed that one line to `"GitHub"`, so v3 started reading and writing `GitHubOAuth Token`. The settings store is an ordinal dictionary, so the old key simply became unreachable, `PersonalAccessToken` fell back to `""`, and `ConfigurationOk` went false - exactly the screenshot in this report.

**Why I would leave it alone.** The value stranded under the old key is a dead credential. v2/v3.0 obtained it through GitHub's OAuth *Authorizations* API using a client id and secret that were hard-coded in this public repository; that API is retired and those credentials were revoked long ago. Current master has dropped the flow entirely and expects a user-pasted Personal Access Token instead. So a migration would take a revoked 2018 token and write it into the live settings store under the guid key, where it would produce 401s from the GitHub calls rather than today's honest "no token configured" state - strictly worse, and it extends the lifetime of a dead secret for no benefit.

A permanent `"Github"` read-fallback would also be the wrong shape: `GitPluginSettingsContainer` is shared by every plugin, so it would re-couple a common component to one plugin's 2018 spelling and partly undo `bc052b7ea`. (The existing `_pluginOldCompatibilityPrefix` fallback there is for the later `{pluginId}.` migration from 2021, which is why v3.0 -> current upgrades do keep the token.)

If anyone does still want it, it should be a plugin-local one-shot copy-and-delete guarded by a migration flag - the pattern already used in `AppSettings` - not a permanent fallback.

**The two secondary points in the report look addressed on master:**

- "it doesn't give any indication that a restart is needed" - `GitHub3Plugin.cs:81` declares "Note: Git Extensions need to be restarted so that the token is taken into account." and surfaces it as a `PseudoSetting` on the settings page.
- no in-app way to obtain a token - there are now "generate token" / "manage token" link labels on that page.

So unless the migration is wanted, this can probably be closed as obsolete: it only affects a settings file last written by GE 2.51.05 or earlier, and the value it would restore no longer works.

**Comment by jrfnl** [comment]
I totally agree this is no longer necessary to be fixed/solved **SEVEN** years after the release of v3 and after it was reported...., especially as multiple majors have been released since....
